src/dialects/sqlite3/index.js

```diff
@@ -18,6 +19,9 @@ var SQLite3_DDL    = require('./schema/ddl')
 
 function Client_SQLite3(config) {
   Client.call(this, config)
+  if (!config.useNullAsDefault) {
```

Comment:
Kind of good idea, but I can't think of any real world situation when one would like to do it. 

I mean that if one decides to use sqlite instead of some other db, it requires modifying configuration at least to set `config.client`. Then one decides that I don't want startup warning, but I still would like that knex throws an error if I try to add multiple columns with partly undefined keys.

That said @rhys-vdw let me know what you think and I'll add another pull request to allow suppress warning by setting `config.useNullAsDefault = false` if it is wanted.